### ISSUE TITLE
[1.12.2] Non consistent (lack of) decoration replacement with different stone generators

### ISSUE THREAD

**Issue by Foreck1** [Aang23/UndergroundBiomesConstructs#240]
There's currently a major limitation when it comes to underground blocks replacement and biomes

In my current setup I'm using [Gave Generator](https://www.curseforge.com/minecraft/mc-mods/cave-generator) to attempt generating "realistic" layers of strata loosely based on the geology of Earth.
Ore stone variants replaces all stone generation as expected, but UBC fails to change other blocks like gravel, sand and overgrown stone to match the corresponding layer, causing a visual disconnection in terrain generation, as well as spawning thematically inconsistent variants of rocks.

![image](https://user-images.githubusercontent.com/47757429/90303731-7f3a6380-de7e-11ea-8198-ea4c60bc8278.png)

in short: Ground layers, gravel and sand are still following the UBC's seeds independently of what stone CG generates.

I'm happy to provide and files or logs necessary to come up with a solution.
Thanks for your time.

**Comment by Foreck1** [comment]
Probably related to https://github.com/Aang23/UndergroundBiomesConstructs/issues/48

**Comment by Morketh** [comment]
Flat world gen has ore pockets
as far as i understand how the flat world gen works i should be looking at a 20 block thick slab of marble sitting on top of 1 block of bedrock. we do infact have a 20 marble thick on 1 bedrock but im not sure the ore is supposed to be included unless specified on the gen parameter string
```3;minecraft:bedrock,20*undergroundbiomes:metamorphic_stone:2```
![image](https://user-images.githubusercontent.com/6291631/90320147-7cca1f00-defb-11ea-8ef0-8b4b434170fe.png)
![image](https://user-images.githubusercontent.com/6291631/90320124-2f4db200-defb-11ea-8a02-168c3a12abaf.png)

**Comment by LemADEC** [comment]
You can't have vanilla or another mod doing the stone placement while UndergroundBiomes does the ore replacement.  
If you want a custom tone generation, then you need to disable the mod in that dimension (just using it to get the block variations) and use another to generate ores while respecting the current stone at that location.  
CoFH world can do that for you, through it's a bit tedious to duplicate for all stone variations.   

In the long term, we could imagine more customization to stone generation in this mod, or adding custom ore generation. None of those are planned at this time.
